ls1intum/Artemis
# `General`: Add option to filter exercises in the statistics which are not included in the score

### Checklist
#### General


#### Client

### Motivation and Context


Some courses use many exercises over the semester to provide their students good learning opportunities. But only the homework counts towards the grade or bonus and the other exercises are not included in the score.
Additionally some students don't participate in the optional exercises or the optional exercises are not graded which leads to a score of 0 points in these exercises.

These two points can make the statistics tab of a course less readable and less informative.

### Description

This PR adds a filter option in the scores tab where students can hide the optional exercises. The UI is the same as for the exercise tab.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Create exercises with different types and different points options (included in the score, bonus, not included) or find a course with these exercises
2. Participate as a student
3. Check the statistic tab correctly displays all graded exercises correctly
4. Check that the filter correctly removes the optional exercises from the scores tab

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





exercise-scores-chart.component.ts: 77% -> 79%
course-statistics.component.ts: 84% -> 86% (also converted from sinon to jest)

### Screenshots


Before filtering:
<img width="1583" alt="Bildschirmfoto 2021-11-25 um 12 50 27" src="https://user-images.githubusercontent.com/38322605/143436771-28ea5099-b4fa-463d-a86b-dbb71cf3ccdf.png">
After filtering:
<img width="1583" alt="Bildschirmfoto 2021-11-25 um 12 50 36" src="https://user-images.githubusercontent.com/38322605/143436748-21af4b2e-4cac-4fe3-8397-d8aaf2f272f6.png">
